include/simdjson/generic/numberparsing.h

```diff
@@ -734,11 +736,12 @@ simdjson_unused simdjson_really_inline simdjson_result<double> parse_double(cons
     if (p-start_exp_digits == 0 || p-start_exp_digits > 19) { return NUMBER_ERROR; }
 
     exponent += exp_neg ? 0-exp : exp;
-    overflow = overflow || exponent < simdjson::internal::smallest_power || exponent > simdjson::internal::largest_power;
   }
 
   if (jsoncharutils::is_not_structural_or_whitespace(*p)) { return NUMBER_ERROR; }
 
+  overflow = overflow || exponent < simdjson::internal::smallest_power || exponent > simdjson::internal::largest_power;
```

Comment:
We have an identical line just above... (line 737). I think it would be better to move it, around near where you have it now...

```C++
    overflow = overflow || exponent < simdjson::internal::smallest_power || exponent > simdjson::internal::largest_power;
```

I think @jkeiser was assuming that without an explicit exponent, you could never exceed the valid range... but it is incorrect.

Reply:
I don't understand what you mean here, I moved this particular line from 737 to 743?

Reply:
If you did do so, then ignore my comment.